Admin migration

jefflau opened the pull request:
I'm going to split the admin migration into functionality and styling so I can get it properly merged with @hiddentao's additions. I have not added a button yet to get to the admin panel. It can be accessed at `/party/address/admin`. The uploader should automatically upload to cloudinary and show up on the edit page.

jefflau commented:
closes https://github.com/wearekickback/app/issues/70

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

makoto commented:
Uploader works very nicely 👍  When we discussed with Jim, he suggested to remove admin functions from /event/event_id page. Would that be covered in the separate PR?

makoto commented on src/api/cloudinary.js:
```diff
@@ -0,0 +1,29 @@
+  const body = new FormData()
+  body.append('file', file)
+  body.append('upload_preset', unsignedUploadPreset)
+  console.log('uploading')
```
Do you need this console.log?

jefflau replied:
Will remove

makoto commented on src/components/SingleEvent/Admin/PartyForm.js:
```diff
@@ -9,30 +64,53 @@ class PartyForm extends Component {
+        })
+      }
+      reader.onabort = () => console.log('file reading was aborted')
+      reader.onerror = () => console.log('file reading has failed')
```
Are they more of `console.warn`?

hiddentao replied:
I wonder if it's worth doing logging via a centralized logging manager, like we do for the server. Would make it easy to capture these logs for cloud storage later on.

jefflau replied:
I can change them to console.warn. I think if we're going to do a centralised logging manager it should be a separate PR. If we did - is there any type of solution you would suggest?

makoto commented:
LGTM though probably better for Ram's sql PR go first then add support for start/end/arriveBy

jefflau commented:
> Uploader works very nicely 👍 When we discussed with Jim, he suggested to remove admin functions from /event/event_id page. Would that be covered in the separate PR?

Yeah I think so. I think it's better to keep them separate. First style this page then => Add admin functionality and remove from single Event page

jefflau commented:
@makoto @hiddentao I've resolved the merge conflicts. Please check this when you have time.

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

hiddentao commented:
<img width="446" alt="screenshot 2019-02-07 at 12 51 26" src="https://user-images.githubusercontent.com/266594/52412598-247cba80-2ad7-11e9-9990-91ae4950eaf3.png">

This section needs cleaning up - Clear should have a an explanation next to it. Ideally it should be in its own section.

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

hiddentao commented:
Wasn't able to change existing event image (https://pr79-app-wearekickback.surge.sh/event/0xed8e57dfdf30be1d942312794ff916caebf1c173/admin/edit). 

Also, shouldn't we allow user to use a URL as well as upload an image?

hiddentao commented:
There needs to be a "back to event" link at the top.

hiddentao commented on src/api/cloudinary.js:
```diff
@@ -0,0 +1,27 @@
+const cloudName = 'dlxrqqprn'
```
Why do we want to do unsigned cloudinary uploads? Why can't we do signed uploads via our server?

jefflau replied:
> There needs to be a "back to event" link at the top.

Will address this in the next PR where I actually add a link to the migration page and style things up

jefflau replied:
I'm fine with trying to address cloudinary uploads via the survey in a separate PR and have it. It was just easier to get it working client-side only and since it was only for event images it doesn't seem like a big deal for now

jefflau commented:
> <img alt="screenshot 2019-02-07 at 12 51 26" width="446" src="https://user-images.githubusercontent.com/266594/52412598-247cba80-2ad7-11e9-9990-91ae4950eaf3.png">
> 
> This section needs cleaning up - Clear should have a an explanation next to it. Ideally it should be in its own section.

I'll address this in the next PR. I'm not dealing with styling here and I'm going to move this into a different section

jefflau commented:
> Wasn't able to change existing event image (https://pr79-app-wearekickback.surge.sh/event/0xed8e57dfdf30be1d942312794ff916caebf1c173/admin/edit).
> 
> Also, shouldn't we allow user to use a URL as well as upload an image?

We could, but if their link breaks then it will show a broken image. I think it might be best to just force them to upload? That way we can deal with how the image is dealt with. We can crop it etc in cloudinary. Unless you are suggesting we take that link and then download the image and then upload it again to cloudinary? But yeah I think for simplicity it, it would be nice to be in control of the image.

jefflau commented:
> Wasn't able to change existing event image (https://pr79-app-wearekickback.surge.sh/event/0xed8e57dfdf30be1d942312794ff916caebf1c173/admin/edit).
> 
> Also, shouldn't we allow user to use a URL as well as upload an image?

I managed to change it on this event: https://pr79-app-wearekickback.surge.sh/event/0x17b8053d11f4f97ce51a20c85390fbf4fdc65cb2

You just need to click on the image. I will add a hover state for the image so it's more obvious you can click and upload a new image

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

jefflau commented:
I'm going to do the image upload in a separate PR as this could be useful for even just Makoto alone to do admin. I've removed mark attended from the normal view. Added a link to the admin view and added mark attended to the list view.

jefflau commented:
I've added image upload and merged it into this branch. It is paired with `image-upload` branch on the server which also has a PR. Please take a look @hiddentao

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

makoto commented:
@jefflau 

At event page

- Please stop showing full name and ETH address

At admin page

- Can you also have link to go back to the event page?

At "participants" tab

- Please add filter by "Status"
- Please add username and twitter handle 
- Please remove T&C and GDPR column (but leave Marketing column)

At "smart contract "tab

- Clicking "Set limit" blows up with "TypeError: Cannot read property 'value' of undefined" error.

The following I assume you address with different PR
- Stop showing clearing period.
- Improve calendar input (I am not sure the clock UI adds anything. I kept clicking the clock assuming that can allow me to specify time, but it does not do that).
- Timezone

jefflau commented:
These should all be done now

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

hiddentao commented:
Todo:

* Fix finalize functionality
* for setLimit - Input textfield onChange error
* for add admins - Input textfield onChange error

makoto commented:
@jefflau let me know when you want me to test again. AFAIK, `Fix finalize functionality` is the only outstanding task.

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh

noblocknoparty-devs commented:
PR deployed at: https://pr79-app-wearekickback.surge.sh
